### ISSUE TITLE
CloudStorageFileSystemProvider.requesterPays() and listBuckets() don't seem to respect retry settings

### ISSUE THREAD

**Issue by lbergelson** [googleapis/java-storage-nio#862]
I'm not sure if this is strictly a bug or more of an incomplete feature.  All of the operations that perform web requests should retry based on the retry configuration settings.

It appears that `CloudStorageFileSystemProvider.requesterPays()` and `CloudStorageFileSystemProvider.listBuckets()` do not perform any retries when making connections.  (except for whatever the internal retry mechanism in storage is).

I don't have strong feelings about `listBuckets()` but we need the `requesterPays` check to be resilient against transient  connection issues. 

#### Environment details

1. gcloud nio
2. OS type and version: all
3. Java version: all
4.  version(s): 0.123.24 and previous

**Comment by BenWhitehead** [comment]
Hi @lbergelson,

Both `listBuckets()` and `get(Bucket)` are idempotent and expected to automatically retry as outlined in https://cloud.google.com/storage/docs/retry-strategy#java which is implemented in `com.google.cloud:google-cloud-storage` from v2.2.0 onward. `com.google.cloud:google-cloud-nio:0.123.17` and onward use `google-cloud-storage` >= 2.2.0.

Can you provide any additional detail about how you're configuring/instantiating your `CloudStorageFileSystemProvider` and `CloudStorageFileSystem`?

**Comment by BenWhitehead** [comment]
I'm going to close due to inactivity, but feel free to reopen if it's still an issue for you.

**Comment by lbergelson** [comment]
@BenWhitehead  Thank you for the clarification and sorry for the confusion.

It looks like I misread the code.  I saw that most of the methods in this library use the `CloudStorageRetryHandler` and that those methods used a different internal retry mechanism. I looked for the link between the CloudStorageConfiguration and the retry strategy used internally but I was looking at the wrong thing because the relevant settings were actually in the StorageOptions.

Sorry for the time waste.
